src/overlays/actors/ovl_En_Guest/z_en_guest.c

```diff
@@ -20,23 +34,200 @@ const ActorInit En_Guest_InitVars = {
+        Actor_ProcessInitChain(&this->actor, initChain);
+
+        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060000F0, NULL, this->limbDrawTable, this->transitionDrawTable,
+                         0x10);
+        gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->objBankIndex].segment);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060042AC, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_060042AC.genericHeader), 0, 0.0f);
+
+        this->actor.draw = &EnGuest_Draw;
+        this->actor.update = &func_80A505CC;
+
+        Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+        Collider_SetCylinder_Set3(globalCtx, &this->colliderCylinder, &this->actor, &colliderInit);
+
+        Actor_SetHeight(&this->actor, 60.0f);
+
+        this->unk_30E = 0;
+        this->unk_30D = 0;
+        this->unk_2CA = 0;
+        this->actor.textId = 0x700D;
+        this->actionFunc = func_80A50518;
+    }
+}
+
+void func_80A5046C(EnGuest* this) {
+    if (this->unk_30D == 0) {
+        if (this->unk_2CA != 0) {
+            this->unk_2CA--;
+        } else {
+            this->unk_30D = 1;
+        }
+    } else {
+        if (this->unk_2CA != 0) {
+            this->unk_2CA--;
+        } else {
+            this->unk_30E += 1;
+            if (this->unk_30E >= 3) {
+                this->unk_30E = 0;
+                this->unk_30D = 0;
+                this->unk_2CA = (s32)Math_Rand_ZeroFloat(60.0f) + 20;
+            } else {
+                this->unk_2CA = 1;
+            }
+        }
+    }
+}
+
+void func_80A50518(EnGuest* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        this->actionFunc = &func_80A5057C;
+    } else if (this->actor.xzDistanceFromLink < 100.0f) {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    }
+}
+
+void func_80A5057C(EnGuest* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 6) {
+        if (func_80106BC8(globalCtx) != 0) {
+            this->actionFunc = &func_80A50518;
+        }
+    }
+}
+
+void func_80A505CC(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+    u32 padding;
+    Player* player;
+
+    player = PLAYER;
+    this->unk_2C8++;
+
+    func_80A5046C(this);
+    this->actionFunc(this, globalCtx);
+
+    this->unk_2A0.unk_18 = player->actor.posRot.pos;
+    if (LINK_IS_ADULT) {
+        this->unk_2A0.unk_14 = 10.0f;
+    } else {
+        this->unk_2A0.unk_14 = 20.0f;
+    }
+    func_80034A14(&this->actor, &this->unk_2A0, 6, 2);
+
+    func_80034F54(globalCtx, this->unk_2CC, this->unk_2EC, 16);
+
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->objBankIndex].segment);
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Actor_SetHeight(&this->actor, 60.0f);
+
+    Collider_CylinderUpdate(&this->actor, &this->colliderCylinder);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder.base);
```

Comment:
this takes a ColliderCylinder, no need to specify base
```suggestion
    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder);
```

Reply:
The function takes a `Collider`, not a `ColliderCylinder`, and `base` is a `Collider` so I used that

Reply:
yeah youre right. we'll need to change that throughout the codebase probably cause we've been giving it just the collider